script/experiments/export_experiment_monitoring_data.py

```diff
@@ -0,0 +1,250 @@
+    required=True,
+    help="Experiment monitoring datasets to be exported",
+    nargs="*",
+    default=[
+        "moz-fx-data-shared-prod.telemetry.experiment_enrollment_daily_active_population",  # noqa E501
+        "moz-fx-data-shared-prod.telemetry.experiment_enrollment_cumulative_population_estimate",  # noqa E501
+    ],
+)
+parser.add_argument(
+    "--date",
+    required=True,
+    help="Export date",
+    type=lambda s: datetime.strptime(s, "%Y-%m-%d"),
+)
+
+
+def get_active_experiments(client, date, dataset):
+    """
+    Determine active experiments.
+
+    Experiments are considered as active if they are currently live or have been live
+    within the past 14 days.
+    Returns the experiment slug and experiment start date.
+    """
+    job = client.query(
+        f"""
+        SELECT DISTINCT experiment, start_date
+        FROM `{dataset}`
+        LEFT JOIN (
+            SELECT
+                start_date,
+                end_date,
+                normandy_slug
+            FROM `moz-fx-data-experiments.monitoring.experimenter_experiments_v1`
+        )
+        ON experiment = normandy_slug
+        WHERE TIMESTAMP_ADD(TIMESTAMP(end_date), INTERVAL 14 DAY) >= TIMESTAMP('{date}')
+        """
+    )
+
+    result = job.result()
+    return [
+        (row.experiment, datetime.combine(row.start_date, datetime.min.time()))
+        for row in result
+    ]
+
+
+def export_data_for_experiment(
+    date, dataset, bucket, gcs_path, source_project, destination_project, experiment
+):
+    """Export the monitoring data for a specific experiment in the dataset."""
+    experiment_slug, start_date = experiment
+    table_name = dataset.split(".")[-1]
+    storage_client = storage.Client(destination_project)
+    client = bigquery.Client(source_project)
+
+    if (date - start_date) > timedelta(days=14):
+        # if the experiment has been running for more than 14 days,
+        # export data as 30 minute intervals
+        query = f"""
+            WITH data AS (
+                SELECT * EXCEPT(experiment) FROM
+                `{dataset}`
+                WHERE experiment = '{experiment_slug}' AND
+                time >= (
+                    SELECT TIMESTAMP(MIN(start_date))
+                    FROM `moz-fx-data-experiments.monitoring.experimenter_experiments_v1`
+                    WHERE normandy_slug = '{experiment_slug}'
+                )
+            )
+            SELECT * EXCEPT(rn) FROM (
+                SELECT *, ROW_NUMBER() OVER (PARTITION BY time) AS rn
+                FROM (
+                    SELECT
+                        * EXCEPT(time),
+                        TIMESTAMP_SECONDS(
+                            UNIX_SECONDS(time) - MOD(UNIX_SECONDS(time), 30 * 60) + 30 * 60
+                            ) AS time,
+                    FROM data
+                    ORDER BY time DESC
+                )
+            )
+            WHERE rn = 1
+            ORDER BY time DESC
+        """
```

Comment:
@emtwo For the exported data, does it make sense to aggregate data into 30 minute intervals if the experiment has been running for 14 days or more? In the first two weeks of the experiment, the exported data will consist of 5 minute intervals.

Reply:
Very nice! I looked at the json files, they look great!
I am mostly thinking outloud here, but I am wondering how we would graph the intervals. E.g. I have a slight concern that 5 min intervals for 2 weeks graphed as-is will be slow on the front-end to graph. I see that Grafana customizes the intervals graphed based on how much data is there. That is possible to do on the front-end if we drop some data points depending on how many there are. 

It's also possible to do it on the ETL side, but I am worried then we cannot retrieve the data if we need it later.

I guess we can start with the 5min intervals for 14 days and 30min for more. If that is an issue, seems like something we can adjust later, right?

Reply:
The underlying experiment_enrollment_aggregates tables aggregate the data in 5 minute intervals. If we'd want to aggregate the data into larger intervals to reduce front-end load, then we can adjust that in the export script easily.